specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-02-01-preview/devices.json

```diff
@@ -0,0 +1,746 @@
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DeviceList": {
+      "description": "List of devices",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "List data",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DeviceModel"
+          }
+        },
+        "nextLink": {
+          "readOnly": true,
+          "type": "string",
+          "description": "When there are too many devices for one page, use this URI to fetch the next page."
+        }
+      }
+    },
+    "DeviceModel": {
+      "type": "object",
+      "description": "Device",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Device properties",
+          "$ref": "#/definitions/DeviceProperties"
+        },
+        "systemData": {
+          "readOnly": true,
+          "type": "object",
+          "description": "Azure Resource Manager metadata containing createdBy and modifiedBy information.",
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ProxyResource"
+        }
+      ]
+    },
+    "DeviceProperties": {
+      "type": "object",
+      "description": "Device properties",
+      "properties": {
+        "schemaVersion": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Version of the device model schema"
+        },
+        "deviceDataSource": {
+          "readOnly": true,
+          "type": "string",
+          "$ref": "#/definitions/DeviceDataSource",
+          "description": "Device data source."
+        },
+        "sensor": {
+          "readOnly": true,
+          "type": "object",
+          "description": "Sensor that scanned this device.",
+          "$ref": "#/definitions/Sensor"
+        },
+        "mergedDevices": {
+          "readOnly": true,
+          "type": "array",
+          "description": "List of merged devices data",
+          "items": {
+            "$ref": "#/definitions/MergedDevice"
+          }
+        },
+        "mergedToDeviceId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The device Id that this device was merged into"
+        },
+        "onboardingStatus": {
+          "readOnly": true,
+          "type": "string",
+          "enum": [
+            "Onboarded",
+            "NotOnboarded",
+            "NotSupported",
+            "InsufficientInfo"
+          ],
+          "x-ms-enum": {
+            "name": "OnboardingStatus",
+            "modelAsString": true
+          },
+          "description": "Device onboarding status."
+        },
+        "name": {
```

Comment:
use "deviceName" or similar.  See the comment in the RPC about not repeating top level properties (even just by name) within the resource properties object:  https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#resource-request-properties-envelope

Reply:
Fixed